### PR TITLE
feat: Add check next angle function

### PR DESCRIPTION
**Pull request by afonsofonbraga**

### CODE REVIEW

**Review by kapeps — APPROVED** [review]

File: test/test_Radar.cpp [hunk 1]
```diff
@@ -54,4 +56,17 @@ TEST_F(RadarTest, it_crashes_with_different_range)
     radar.addEcho(range, sweep_length, step_size, heading_0, echo);
     ASSERT_THROW(radar.addEcho(range + 2.0, sweep_length, step_size, heading_1, echo),
         std::runtime_error);
+}
+
+TEST_F(RadarTest, verify_next_angle_works_as_expected)
```
Comment (doudou) [inline test/test_Radar.cpp]: "works as expected" is not a specification. You need to split your tests and say exactly what behavior you are testing against.

TEST_F(..., it_returns_true_for_the_first_echo) {
}

TEST_F(..., it_returns_true_if_the_angle_is_within_size_times_step_from_the_first_with_a_positive_step_angle) {
}

TEST_F(..., it_returns_true_if_the_angle_is_within_size_times_step_from_the_first_with_a_negative_step_angle) {
}
```

File: src/Radar.cpp [hunk 2]
```diff
@@ -26,42 +27,49 @@ Radar::~Radar()
...
     }
     else {
-        base::Angle start = heading - step_angle * sweep_timestamps.size();
-        if (!this->start_heading.isApprox(start)) {
-            throw std::runtime_error(
-                "Current Start heading differs from configured values! Configured: " +
-                to_string(this->start_heading.getRad()) +
-                " Current:" + to_string(start.getRad()));
+        Angle expected = start_heading + step_angle * sweep_timestamps.size();
+        if (!expected.isApprox(heading)) {
```
Comment (doudou) [inline src/Radar.cpp]: Shouldn't that be ... verifyNextAngle ?

Comment (DeboraTahara) [inline src/Radar.cpp]: Why don't you use verifyNextAngle here?

File: src/Radar.cpp [hunk 3]
```diff
@@ -48,17 +48,18 @@ void Radar::addEcho(float range,
...
+                            " Current:" + to_string(step_angle.getRad()));
     }
     else {
-        Angle expected = start_heading + step_angle * sweep_timestamps.size();
-        if (!expected.isApprox(heading)) {
-            throw runtime_error("Current angle differs from expected! Expected: " +
-                                to_string(expected.getRad()) +
-                                " Current:" + to_string(heading.getRad()));
+        if (!verifyNextAngle(heading)) {
+            Angle expected = start_heading + step_angle * sweep_timestamps.size();
```
Comment (DeboraTahara) [inline src/Radar.cpp]: you forgot it here!

**Review by DeboraTahara — APPROVED** [review]

File: src/Radar.cpp (hunk 3 above)
Comment (afonsofonbraga) [reply src/Radar.cpp]: my bad, sorry.

**Review by doudou — APPROVED** [review]

**Pull request merged**